kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -1021,48 +1037,72 @@ protected void handleOffsetFetchRequest(KafkaHeaderAndRequest offsetFetch,
         checkState(groupCoordinator != null,
             "Group Coordinator not started");
 
-        Map<TopicPartition, OffsetFetchResponse.PartitionData> unknownPartitions = new HashMap<>();
+        CompletableFuture<List<TopicPartition>> authorizeFuture = new CompletableFuture<>();
 
         // replace
         Map<TopicPartition, TopicPartition> replacingIndex = new HashMap<>();
-        KeyValue<Errors, Map<TopicPartition, OffsetFetchResponse.PartitionData>> keyValue =
-            groupCoordinator.handleFetchOffsets(
-                    request.groupId(),
-                    request.partitions() == null ? Optional.ofNullable(request.partitions()) : Optional.of(
-                            request.partitions()
-                            .stream()
-                            .map(tp -> {
-                                try {
-                                    TopicPartition newTopicPartition = new TopicPartition(
-                                            new KopTopic(tp.topic()).getFullName(), tp.partition());
-                                    replacingIndex.put(newTopicPartition, tp);
-                                    return newTopicPartition;
-                                } catch (KoPTopicException e) {
-                                    log.warn("Invalid topic name: {}", tp.topic(), e);
-                                    unknownPartitions.put(tp, OffsetFetchResponse.UNKNOWN_PARTITION);
-                                    return null;
-                                }
-                            })
-                            .filter(Objects::nonNull)
-                            .collect(Collectors.toList()))
-            );
 
-        if (log.isDebugEnabled()) {
-            log.debug("OFFSET_FETCH Unknown partitions: {}", unknownPartitions);
-        }
+        List<TopicPartition> authorizedPartitions = new ArrayList<>();
+        Map<TopicPartition, OffsetFetchResponse.PartitionData> unauthorizedPartitionData = new HashMap<>();
 
-        if (log.isTraceEnabled()) {
-            StringBuffer traceInfo = new StringBuffer();
-            replacingIndex.forEach((inner, outer) ->
-                    traceInfo.append(String.format("\tinnerName:%s, outerName:%s%n", inner, outer)));
-            log.trace("OFFSET_FETCH TopicPartition relations: \n{}", traceInfo.toString());
+        if (request.partitions() == null || request.partitions().isEmpty()) {
+            authorizeFuture.complete(null);
+        } else {
+            AtomicInteger partitionCount = new AtomicInteger(request.partitions().size());
+
+            Runnable completeOneAuthorization = () -> {
+                if (partitionCount.decrementAndGet() == 0) {
+                    authorizeFuture.complete(authorizedPartitions);
+                }
+            };
+            request.partitions().forEach(tp -> {
+                String fullName =  new KopTopic(tp.topic()).getFullName();
```

Comment:
Why did you remove the check for `KoPTopicException` that was introduced from #585?